src/Compilers/Core/Portable/CodeGen/ReferenceDependencyWalker.cs

```diff
@@ -145,6 +145,25 @@ private static void VisitMethodReference(Cci.IMethodReference methodReference, E
             }
         }
 
+        internal static void VisitSignature(Cci.ISignature signature, EmitContext context)
```

Comment:
>VisitSignature [](start = 29, length = 14)

It feels like we should be calling this method for function pointer types in VisitTypeReference method, otherwise we won't record references in declarations.